### PR TITLE
update precompiled to use function-selector

### PR DESCRIPTION
**Pull request by zjb0807**
Closes: #973

### CODE REVIEW

**Review by ermalkaleci — APPROVED** [review]

**Review by xlc — APPROVED** [review]

File: primitives/src/lib.rs [hunk 1]
```diff
@@ -190,3 +194,13 @@ pub const H160_PREFIX_TOKEN: [u8; 19] = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
 pub const H160_PREFIX_DEXSHARE: [u8; 12] = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1];
 
 pub type NFTBalance = u128;
+
+pub fn get_function_selector(s: &str) -> [u8; 4] {
```
[the remark this answers is not in the document]
Comment (zjb0807) [reply primitives/src/lib.rs:198]: Both `evm-bridge` and `runtime-common` use it. So put it here

[the remark this answers is not in the document]
Comment (zjb0807) [reply primitives/src/lib.rs:198]: I'll move it to mock.rs

**Review by xlc — APPROVED** [review]

**Review by shaunxw — APPROVED** [review]

File: runtime/common/src/precompile/dex.rs [hunk 2]
```diff
@@ -38,17 +38,17 @@ pub struct DexPrecompile<AccountId, AddressMapping, CurrencyIdMapping, Dex>(
...
+#[repr(u32)]
+pub enum Action {
+	GetLiquidityPool = 0xf4f31ede,
+	GetLiquidityTokenAddress = 0xffd73c4a,
+	GetSwapTargetAmount = 0x4d60beb1,
+	GetSwapSupplyAmount = 0xdbcd19a2,
+	SwapWithExactSupply = 0x579baa18,
+	SwapWithExactTarget = 0x9782ac81,
+	AddLiquidity = 0x4ea5efef,
+	RemoveLiquidity = 0xda613b51,
```
Comment (shaunxw) [inline runtime/common/src/precompile/dex.rs:51]: IMO actions variant values should be generated automatically in the future improvement. Manual adding them could be error-prune.

Comment (xlc) [reply runtime/common/src/precompile/dex.rs:51]: we have unit test to assert those value are correct so it is not too bad. but yes I think we can have some macro to generate them

**Pull request merged**